### PR TITLE
[IMP] accounting: update thailand localisations for PromptPay

### PR DESCRIPTION
**Pull request by tong-odoo**
Impacted Version:

- 17.0 and above

This commit improve below features:

- PromptPay QR code documentation

task-3506522

### CODE REVIEW

**Review by chiaraprattico — CHANGES_REQUESTED** [review]

**Comment by tong-odoo** [comment]
@chiaraprattico Thanks for the review. I have fixed the comments above. Please take a look! :smile:

File: content/applications/finance/fiscal_localizations/thailand.rst [hunk 1]
```diff
@@ -109,3 +109,57 @@ in the app, open the **contact form** of the company and under the :guilabel:`Sa
...
+
+Go to :menuselection:`Accounting --> Configuration --> Settings`. Under the :guilabel:`Customer
+Payments` section, activate the :guilabel:`QR Codes` feature.
+
+PromptPay QR bank account configuration
+---------------------------------------
+
+Go to :menuselection:`Contacts --> Configuration --> Bank Accounts` and select the bank account for
+which you want to activate PromptPay QR. Set the :guilabel:`Proxy Type` and fill in the :guilabel:
+`Proxy Value` field depending on the chosen type.
```
Comment (chiaraprattico) [inline content/applications/finance/fiscal_localizations/thailand.rst:135]: the guilabel Proxy Value is not working

**Review by chiaraprattico — COMMENTED** [review]
Hi @tong-odoo just one guilabel is not working, but I'm sending it over already to my colleague for finalizing and pushing it.

**Comment by tong-odoo** [comment]
@chiaraprattico Thanks for checking! I have fixed the guilabel as well. Please take a look :smile:

File: content/applications/finance/fiscal_localizations/thailand.rst [hunk 2]
```diff
@@ -109,3 +109,57 @@ in the app, open the **contact form** of the company and under the :guilabel:`Sa
...
+---------------------------------------
+
+Go to :menuselection:`Contacts --> Configuration --> Bank Accounts` and select the bank account for
+which you want to activate PromptPay QR. Set the :guilabel:`Proxy Type` and fill in the
+:guilabel:`Proxy Value` field depending on the chosen type.
+
+.. important::
+   - The :guilabel:`Include Reference` checkbox doesn't work for PromptPay QR codes.
+
+.. image:: thailand/thailand-promptpay-bank-setting.png
```
Comment (jcs-odoo) [inline content/applications/finance/fiscal_localizations/thailand.rst:140]: No need for the prefix "thailand" in the filename, as it is already in the path with the name of the folder. Rather, I suggest a prefix about the topic
```suggestion
.. image:: thailand/qr-promptpay-bank.png
```

File: content/applications/finance/fiscal_localizations/thailand.rst [hunk 3]
```diff
@@ -109,3 +109,57 @@ in the app, open the **contact form** of the company and under the :guilabel:`Sa
...
+.. seealso::
+   :doc:`../accounting/bank`
+
+Bank journal configuration
+--------------------------
+
+Go to :menuselection:`Accounting --> Configuration --> Journals`, open the bank journal, then fill
+out the :guilabel:`Account Number` and :guilabel:`Bank` under the :guilabel:`Journal Entries` tab.
+
+.. image:: thailand/thailand-bank-account-journal-setting.png
```
Comment (jcs-odoo) [inline content/applications/finance/fiscal_localizations/thailand.rst:152]: For example here, this name could apply to other topics than the QR code section.
```suggestion
.. image:: thailand/qr-bank-journal.png
```

File: content/applications/finance/fiscal_localizations/thailand.rst [hunk 4]
```diff
@@ -109,3 +109,57 @@ in the app, open the **contact form** of the company and under the :guilabel:`Sa
...
+   :alt: Bank Account's journal configuration
+
+Issue invoices with PromptPay QR code
+-------------------------------------
+
+When creating a new invoice, open the :guilabel:`Other Info` tab and set the :guilabel:`Payment
+QR-code` option to :guilabel:`EMV Merchant-Presented QR-code`.
+
+.. image:: thailand/thailand-qr-code-invoice-setting.png
```
Comment (jcs-odoo) [inline content/applications/finance/fiscal_localizations/thailand.rst:161]:
```suggestion
.. image:: thailand/qr-code-invoice-emv.png
```

File: content/applications/finance/fiscal_localizations/thailand.rst [hunk 5]
```diff
@@ -109,3 +109,57 @@ in the app, open the **contact form** of the company and under the :guilabel:`Sa
 
 .. tip::
    This information is used in the **tax invoice** PDF report and PND **tax report** export.
+
+PromptPay QR code on invoices
+=============================
+
+The **PromptPay QR code** is a QR code that can be added to invoices to allow customers to pay
+their bills using the PromptPay-supported bank mobile application. The QR code is generated based on
+the **invoice amount** and one of the following **merchant information**:
```
Comment (jcs-odoo) [inline content/applications/finance/fiscal_localizations/thailand.rst:118]:
```suggestion
The **PromptPay QR code** is a QR code that can be added to invoices to allow customers to pay their
bills using the PromptPay-supported bank mobile application. The QR code is generated based on the
**invoice amount** and one of the following **merchant information**:
```

**Review by jcs-odoo — CHANGES_REQUESTED** [review]
Hello @tong-odoo 

Can you update the file names? See my suggestions below. I'll merge this afterward :)

Thanks!

**Comment by tong-odoo** [comment]
@jcs-odoo Thanks for the review! I have updated the file names and your suggestions. Please take a look. :smile:

**Review by jcs-odoo — APPROVED** [review]
Thanks @tong-odoo 

Usually, it's best not to mix different types of changes if they can be applied to older versions as well (here --> changing existing filenames)
Otherwise, it becomes harder to maintain the doc across different versions.

But no worries, I created another PR to backport these changes to older versions of the doc :)
https://github.com/odoo/documentation/pull/5891

Thank you for your work!

@robodoo r+

**Comment by tong-odoo** [comment]
> Thanks @tong-odoo
> 
> Usually, it's best not to mix different types of changes if they can be applied to older versions as well (here --> changing existing filenames) Otherwise, it becomes harder to maintain the doc across different versions.
> 
> But no worries, I created another PR to backport these changes to older versions of the doc :) #5891
> 
> Thank you for your work!
> 
> @robodoo r+

Sure thank you!

**Pull request closed without merging**